File: backend/GameApp.Application/UseCases/ItemUseCases/ItemGetUseCase.cs
using GameApp.Domain.Entities.Items;
using GameApp.Domain.ValueObjects.Items;

namespace GameApp.Application.UseCases.ItemUseCases;

public interface ItemGetUseCase
{
    public Task<Item?> GetItem(Guid id);

    public Task<Item?> GetItemByName(ItemName name);

    public Task<IEnumerable<Item>> GetAllItems();
}

File: frontend/gameapp-frontend/src/application/usecases/item-use-cases/item-get-use-case.ts
import { Item } from '../../../domain/entities/items/item';
import type { ItemRarity } from '../../../domain/enumerates/item-rarity';
import type { ItemName } from '../../../domain/value-objects/items/item-name';
import type { ItemType } from '../../enumerates/item-type';

export interface IItemGetUseCase {
  getItem(id: string): Promise<Item>;
  getItemByName(name: ItemName): Promise<Item>;
  getItemByFilter(type?: ItemType, rarity?: ItemRarity): Promise<Item[]>;
  getAllItems(): Promise<Item[]>;
}
